backend/main-service/src/main/java/kkakka/mainservice/cart/application/CartService.java

```diff
@@ -29,32 +33,60 @@ public CartService(CartRepository cartRepository, CartItemRepository cartItemRep
         this.memberRepository = memberRepository;
     }
 
-    public String saveCartItem(CartRequestDto cartRequestDto) {
+    @Transactional
+    public Long saveOrUpdateCartItem(CartRequestDto cartRequestDto) {
 
         /* 테스트 데이터 */
         Optional<Member> member = memberRepository.findById(cartRequestDto.getMemberId());
 
-        /* 로그인 상태 체크 */
+        Cart cart = findOrCreateCart(member.get());
+        Optional<CartItem> cartItem = Optional.ofNullable(
+                cartItemRepository.findByMemberIdandProductId(member.get().getId(), cartRequestDto.getProductId())
+        );
 
-        /* 장바구니 존재 유무 체크 ex) 장바구니에 담아둔 아이템 있는지 */
-        Cart cart = new Cart();
-        cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
-        if (cart == null) {
-            cartRepository.save(new Cart(member.get()));
-            cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
+        // 동일 상품 이미 있으면 수량 업데이트
+        if (cartItem.isPresent()) {
+            Integer preQuantiry = cartItem.get().getQuantity();
+            cartItem.get().setQuantity(preQuantiry + cartRequestDto.getQuantity());
+            cartItemRepository.save(cartItem.get());
+            return member.get().getId();
         }
 
-        /* 현재 장바구니에 동일한 상품 있는지 체크 */
-        CartItem item = cartItemRepository.findByMemberIdandProductId(member.get().getId(), cartRequestDto.getProductId()); // Test용 Member
-//        System.out.println(item.toString());
-
         /* 장바구니 아이템 추가 */
         Optional<Product> product = productRepository.findById(cartRequestDto.getProductId());
-        cartItemRepository.save(new CartItem(cart, product.get(), cartRequestDto.getQuantity()));
+        cartItem = Optional.ofNullable(new CartItem(cart, product.get(), cartRequestDto.getQuantity()));
+        cartItemRepository.save(cartItem.get());
+
+        return member.get().getId();
+    }
+
+    /* 멤버 장바구니 목록 조회 */
+    public List<CartResponseDto> findAllCartItemByMember(Member member) {
+
+        List<CartItem> cartItemList = cartItemRepository.findAllByMemberId(member.getId());
+
+        List<CartResponseDto> result = new ArrayList<>();
+        cartItemList.forEach(c -> {
+            result.add(CartResponseDto.from(c));
+        });
 
-        return "";
+        return result;
     }
 
+    /* 장바구니 아이템 삭제 */
+    @Transactional
+    public boolean deleteCartItem(Long cartItemId) {
 
+        cartItemRepository.deleteCartItemById(cartItemId);
+        return true;
+    }
+
+    public Cart findOrCreateCart(Member member) {
+        return cartRepository.findByMemberId(member.getId()).orElseGet(() -> createCart(member));
```

Comment:
```suggestion
        return cartRepository.findByMemberId(member.getId())
                      .orElseGet(() -> createCart(member));
```

저는 이런거 개행 한번 해주면 가독성이 훨 좋아보이더라구요!

Reply:
반영하겟습니다 ~!